src/Compression/Compression.cs

```diff
@@ -300,11 +300,19 @@ private static void ThrowCannotWriteToDeflateStreamException()
 			throw new InvalidOperationException ("Writing to the compression stream is not supported.");
 		}
 
+#if !DOTNET_TODO
+		// DOTNET_TODO: TaskToApm is internal API in mscorlib, so we can't use it.
+		// Although I'm not sure if we even need to override these methods,
+		// the base class implementation works (but synchronously),
+		// and the documentation recommends using ReadAsync instead.
+		// If we decide to implement this, we could just copy the TaskToApm implementation:
+		// https://github.com/microsoft/referencesource/blob/a7bd3242bd7732dec4aebb21fbc0f6de61c2545e/mscorlib/system/threading/Tasks/TaskToApm.cs
 		public override IAsyncResult BeginRead (byte[] buffer, int offset, int count, AsyncCallback asyncCallback, object asyncState) =>
 			TaskToApm.Begin(ReadAsync (buffer, offset, count, CancellationToken.None), asyncCallback, asyncState);
 
 		public override int EndRead (IAsyncResult asyncResult) =>
 			TaskToApm.End<int> (asyncResult);
+#endif // !DOTNET_TODO
```

Comment:
So, this was a similar implementation to what it was done back in the day in the compression implementations of mono for zips. The APIs is not very used. If the base class implementation works (sync) is not a big issues since they should be using the Task version. I think it might be a good opportunity to get this out as outdated code.

Reply:
It is actually problematic. Sorry I didn't notice your comment and started [new thread below](https://github.com/xamarin/xamarin-macios/pull/8070#discussion_r390585562).